src/run/headers.ts

```diff
@@ -242,11 +250,25 @@ export const setCacheControlHeaders = (
       .log('NetlifyHeadersHandler.trailingSlashRedirect')
   }
 
-  if (status === 404 && request.url.endsWith('.php')) {
-    // temporary CDN Cache Control handling for bot probes on PHP files
-    // https://linear.app/netlify/issue/FRB-1344/prevent-excessive-ssr-invocations-due-to-404-routes
-    headers.set('cache-control', 'public, max-age=0, must-revalidate')
-    headers.set('netlify-cdn-cache-control', `max-age=31536000, durable`)
+  if (status === 404) {
+    if (request.url.endsWith('.php')) {
+      // temporary CDN Cache Control handling for bot probes on PHP files
+      // https://linear.app/netlify/issue/FRB-1344/prevent-excessive-ssr-invocations-due-to-404-routes
+      headers.set('cache-control', 'public, max-age=0, must-revalidate')
+      headers.set('netlify-cdn-cache-control', `max-age=31536000, durable`)
+      return
+    }
+
+    if (
+      typeof requestContext.pageHandlerRevalidate !== 'undefined' &&
+      ['GET', 'HEAD'].includes(request.method) &&
+      !headers.has('cdn-cache-control') &&
+      !headers.has('netlify-cdn-cache-control')
+    ) {
+      // handle CDN Cache Control on 404 Page responses
+      setCacheControlFromRequestContext(headers, requestContext.pageHandlerRevalidate)
+      return
+    }
```

Comment:
Just making sure I understand correctly - this will only add cache control headers to 404 pages with a revalidate value. What about 404 pages that use getStaticProps but don't specify revalidate?

Reply:
The method (I extracted from existing handling for App Router Route Handlers) supports `false | number` for `revalidate`:
https://github.com/opennextjs/opennextjs-netlify/blob/330dca0ceee048d05dbed4958e3c33e1008c3ae0/src/run/headers.ts#L220

`false` meaning that `revalidate` is not specified and then it would set ~permanent / 1 year caching header

Reply:
The test adjustment I did (minus currently wrong "message") - https://github.com/opennextjs/opennextjs-netlify/blob/330dca0ceee048d05dbed4958e3c33e1008c3ae0/tests/integration/page-router.test.ts#L117-L132 is for the case of `revalidate` not being specified and test asserting 1 year caching headers - this is used test fixture for it https://github.com/opennextjs/opennextjs-netlify/blob/330dca0ceee048d05dbed4958e3c33e1008c3ae0/tests/fixtures/page-router-base-path-i18n/pages/404.js#L9-L16

We currently don't have test case for 404 with `getStaticProps` + `revalidate` being actually defined - I did some quick manual test and it _seemed_ to be working (but this was real quick test and with rushing it maybe it's not). We should add a test case for it